compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -237,36 +237,34 @@ convertRationalDecimal env num denom
     maxPrecision = 10 :: Integer
 
 -- | Convert a rational number into a fixed scale Numeric literal. We check
--- that the scale is in bounds, and the number can be represented without
+-- that the scale is in bound, and the number can be represented without
 -- overflow or loss of precision.
 convertRationalNumericMono :: Env -> Integer -> Integer -> Integer -> ConvertM LF.Expr
-convertRationalNumericMono env scale num denom =
-    if  | scale < 0 || scale > 37 ->
-            unsupported
-                ("Tried to construct value of type Numeric " ++ show scale ++ ", but scale is out of bounds. Scale must be between 0 through 37, not " ++ show scale ++ ".")
-                scale
-
-        | abs (r * 10 ^ scale) >= upperBound128Bit ->
-            unsupported
-                ("Rational is out of bounds: " ++
-                show ((fromInteger num / fromInteger denom) :: Double) ++
-                ". The range of values representable by the Numeric " ++ show scale ++ " type is  -10^" ++ show (38-scale) ++ " + 1  through  10^" ++ show (38-scale) ++ " - 1.")
-                (num, denom)
+convertRationalNumericMono env scale num denom
+    | scale < 0 || scale > fromIntegral numericMaxScale =
+        unsupported
+            ("Tried to construct value of type Numeric " ++ show scale ++ ", but scale is out of bounds. Scale must be between 0 through 37, not " ++ show scale ++ ".")
+            scale
+
+    | abs (rational * 10 ^ scale) >= 10 ^ numericMaxPrecision =
+        unsupported
+            ("Rational is out of bounds: " ++ show double ++ ". The range of values representable by the Numeric " ++ show scale ++ " type is  -10^" ++ show maxPower ++ " + 1  through  10^" ++ show maxPower ++ " - 1.")
```

Comment:
Shouldn't this be something like
```suggestion
            ("Rational is out of bounds: " ++ show double ++ ". The range of values representable by the Numeric " ++ show scale ++ " type is  -10^" ++ show maxPower ++ " + 10^-" ++ show scale ++ "  through  10^" ++ show maxPower ++ " - 1.")
```

Reply:
ah, you're right

Reply:
gosh, what a cumbersome number to describe ... perhaps I can say "Numeric scale can only represent numbers greater than -10^maxPower and smaller than 10^maxPower"

Reply:
I would use language like "between ... (exlusive)" or "strictly between". But I don't really care.